packages/KInlineEdit/KInlineEdit.vue

```diff
@@ -0,0 +1,124 @@
+<template>
+  <div
+    id="editable-wrapper"
+    class="k-inline-edit">
+    <input
```

Comment:
How does a user declare that an input is required? For example, you can't have an empty service package name, so are we going to be able to handle that?

Reply:
right now you can't so open to suggestions. Could do something like autosuggest with `inputProps` and bind the object. Thoughts?

Reply:
another option could be if you pass a required true prop we could disallow empty inputs. And if you remove all text and click away it negates your change and doesn't make any updates.

Reply:
@travispamaral is there even a use case for removing a value entirely? I think for the first iteration we can assume it's required and reset the value to the original, but still emit an event of empty text so that users (aka "us") could alert in some way that it's required e.g. toaster/kalert/red border or something. We can address what that looks like in implementation pr. Interestingly, if "empty" == "reset to original", then we have a new feature of reseting the input.

Reply:
IDK if sending empty makes sense. We don't "reset" or even manage the text internally. The text is always passed up and then reassigned via the element in the slot. To me I feel the implementation should handle the rules. 

```js
// handler in implementing component
handleEmitChange(val) {
  if (val.length === 0) {
    $toaster.notify({message: "Can't be empty", type: "danger" })
  } else {
    this.text = val
  }
}

Reply:
Another option is we could add a required function and if met emits/changes input to red. Then in the parent you can do whatever you want if the conditions are met.

Reply:
yea after playing with it a little bit I see what you mean, that we should make the example check the truthiness of the updated value. Here's an example:

```vue
<Komponent :data="{ inlineText: 'Click to edit me' }" v-slot="{ data }">
  <div>
    Updated: {{ data.inlineText }}
    <KInlineEdit @changed="newVal => { if(newVal.length) { data.inlineText = newVal } }">
      <h3>{{ data.inlineText }}</h3>
    </KInlineEdit>
  </div>
</Komponent>
```